packages/palette/src/elements/Tabs/Tabs.tsx

```diff
@@ -1,278 +1,87 @@
-import React, { Ref } from "react"
-import styled, { css } from "styled-components"
-import { JustifyContentProps, WidthProps } from "styled-system"
-import { Box } from "../Box"
-import { Flex, FlexProps } from "../Flex"
-import { Join } from "../Join"
-import { Sans } from "../Typography"
-
-import { color, media } from "../../helpers"
-import { Text } from "../Text"
+import React, { useState } from "react"
+import { flattenChildren } from "../../helpers/flattenChildren"
+import { useUpdateEffect } from "../../utils/useUpdateEffect"
+import { BaseTabs, BaseTabsProps } from "../BaseTabs"
+import { TabProps } from "./Tab"
+import { TabsTab, TabsTabProps } from "./TabsTab"
```

Comment:
And necessary to have a `TabsTab` as well as a `Tab`?

Reply:
If all of this division is strictly necessary then 👍 but it be a little more straight forward if there was a `Tabs` component, and inside it was a `Tabs` and `Tab`

Reply:
But yeah; TabsTab doesn't need to be exported from the index.

Reply:
![](https://static.damonzucconi.com/_capture/q25SR9Ut.png)

Reply:
Of course, I don't mean to dismiss this work! I suspect that we're both on the same page about where this has been and is going...

Reply:
> I don't understand why this component is so complex given the tiny API.

Two reasons:

* Shovelling props into <Tabs> using the <Tab> component. This is where you get the problem with the two different tabs, one visual/clickable; the other just part of the Tabs API.
* `Stepper` — it's close enough to Tabs that Tabs were used to build it but it requires disabling and altering the tab buttons, which is *weird*.